llvm/lib/Target/X86/X86ISelLowering.cpp

```diff
@@ -28211,9 +28212,14 @@ static SDValue LowerABS(SDValue Op, const X86Subtarget &Subtarget,
   MVT VT = Op.getSimpleValueType();
   SDLoc DL(Op);
 
+  // Since X86 does not have CMOV for 8-bit integer, promote to 32-bit.
+  if (VT == MVT::i8)
```

Comment:
Should we limit to 64-bit only? I assume we will reduce the available registers by promotion.